### PR TITLE
adapt to new pool configuration API in Vert.x 4.1

### PR DESCRIPTION
**Pull request by gavinking**

### CODE REVIEW

**Comment by gavinking** [comment]
(Of course this patch is _not_ compatible with Vert.x 4.0.)

**Comment by gavinking** [comment]
@Davide @Sanne this changes are actually pretty important to take advantage of the new features of the pool in 4.1. (And users with the famous idle connection timeout issues.)

**Comment by Sanne** [comment]
I'm re-running all CI jobs as we just upgraded

File: documentation/src/main/asciidoc/reference/introduction.adoc [hunk 1]
```diff
@@ -912,8 +912,9 @@ following configuration properties:
 | Configuration property name                          | Purpose
 
 | `hibernate.vertx.pool.max_wait_queue_size`           | The maximum connection requests allowed in the wait queue
-| `hibernate.vertx.pool.connect_timeout`               | The maximum time to wait when opening a new connection, in milliseconds
-| `hibernate.vertx.pool.idle_timeout`                  | The maximum time a connection may sit idle, in seconds
+| `hibernate.vertx.pool.connect_timeout`               | The maximum time to wait when requesting a pooled connection, in milliseconds
```
Comment (gavinking) [inline documentation/src/main/asciidoc/reference/introduction.adoc:915]: Notice the semantic change here.

File: documentation/src/main/asciidoc/reference/introduction.adoc [hunk 2]
```diff
@@ -912,8 +912,9 @@ following configuration properties:
...
+| `hibernate.vertx.pool.idle_timeout`                  | The maximum time a connection may sit idle, in milliseconds
```
Comment (gavinking) [inline documentation/src/main/asciidoc/reference/introduction.adoc:916]: Notice the semantic change here.

**Comment by gavinking** [comment]
I've updated the docs. Let me know when I can merge this.

**Review by DavideD — APPROVED** [review]

**Comment by DavideD** [comment]
LGTM

**Pull request merged**